dlt-database/migrations/0003-refactor_transaction_recipe.ts

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+export async function upgrade(queryFn: (query: string, values?: any[]) => Promise<Array<any>>) {
+  // write upgrade logic as parameter of queryFn
+  await queryFn(`DROP TABLE \`confirmed_transactions\`;`)
+  await queryFn(`DROP TABLE \`transaction_recipes\`;`)
+  
+  await queryFn(
+    `ALTER TABLE \`accounts\` MODIFY COLUMN  \`derivation_index\` int(10) unsigned NULL DEFAULT NULL;`,
+  )
+  await queryFn(
+    `ALTER TABLE \`accounts\` ADD COLUMN \`account_balance_created_at\` decimal(40,20) NOT NULL DEFAULT 0 AFTER \`balance_date\`;`,
+  )
+  await queryFn(
+    `ALTER TABLE \`accounts\` ADD COLUMN \`balance_created_at_date\` datetime(3) NOT NULL DEFAULT CURRENT_TIMESTAMP(3) AFTER \`account_balance_created_at\`;`,
+  )
+
+  await queryFn(
+    `CREATE TABLE \`transactions\` (
+        \`id\` bigint unsigned NOT NULL AUTO_INCREMENT,
+        \`iota_message_id\` varbinary(32) NULL DEFAULT NULL,
+        \`backend_transaction_id\` bigint unsigned NULL DEFAULT NULL,
+        \`paring_transaction_id\` bigint unsigned NULL DEFAULT NULL,
+        \`signing_account_id\` int unsigned NULL DEFAULT NULL,
+        \`recipient_account_id\` int unsigned NULL DEFAULT NULL,
+        \`sender_community_id\` int unsigned NOT NULL,
+        \`recipient_community_id\` int unsigned NULL DEFAULT NULL,
+        \`amount\` decimal(40, 20) NULL DEFAULT NULL,
+        \`account_balance_created_at\` decimal(40, 20) NOT NULL,
```

Comment:
the naming-pattern `ending-with: ...created_at` for a column of type `decimal` is not quite good.